tests/phpunit/tests/option/option.php

```diff
@@ -118,6 +118,42 @@ public function test_get_option_notoptions_cache() {
 		$this->assertSame( 0, $after - $before );
 	}
 
+	/**
+	 * @ticket 58277
+	 *
+	 * @covers ::get_option
+	 */
+	public function test_get_option_notoptions_set_cache() {
+		get_option( 'invalid' );
+
+		$before = get_num_queries();
+		$value  = get_option( 'invalid' );
+		$after  = get_num_queries();
+
+		$notoptions = wp_cache_get( 'notoptions', 'options' );
+
+		$this->assertSame( 0, $after - $before, 'No more queries should be run on second run on function' );
+		$this->assertArrayHasKey( 'invalid', $notoptions, 'Check that option name is in notoptions cache' );
+	}
+
+	/**
+	 * @ticket 58277
+	 *
+	 * @covers ::get_option
+	 */
+	public function test_get_option_notoptions_do_not_load_cache() {
+		add_option( 'foo', 'bar', '', 'no' );
+		wp_cache_delete( 'notoptions', 'options' );
+
+		$before = get_num_queries();
+		$value  = get_option( 'foo' );
+		$after  = get_num_queries();
+
+		$notoptions = wp_cache_get( 'notoptions', 'options' );
+
+		$this->assertSame( 0, $after - $before, 'No more queries should be run on second run on function' );
+		$this->assertFalse( $notoptions, 'Check notoptions cache is not set' );
```

Comment:
```suggestion
		$this->assertSame( 0, $after - $before, 'The options cache was not hit on the second call to `get_option()`.' );
		$this->assertFalse( $notoptions, 'The notoptions cache should not be set.' );
```
Some improvements to the assertion failure messages.